compiler/src/iree/compiler/Codegen/Common/test/reconcile_translation_info.mlir

```diff
@@ -431,3 +470,77 @@ hal.executable private @no_loop_default_workgroup_count {
+      %x, %y, %z = iree_tensor_ext.dispatch.workgroup_count_from_slice %arg1, %arg2, %arg3
+      hal.return %x, %y, %z : index, index, index
+    }
+    builtin.module {
+      func.func @entry_point0() {
+        %cst0 = hal.interface.constant.load layout(#pipeline_layout) ordinal(0) : index
+        %cst1 = hal.interface.constant.load layout(#pipeline_layout) ordinal(1) : index
+        %cst2 = hal.interface.constant.load layout(#pipeline_layout) ordinal(2) : index
+        %0 = iree_tensor_ext.dispatch.workload.ordinal %cst0, 0 : index
+        %1 = iree_tensor_ext.dispatch.workload.ordinal %cst1, 1 : index
+        %2 = iree_tensor_ext.dispatch.workload.ordinal %cst2, 2 : index
+        scf.forall (%arg0, %arg1) = (0, 0) to (%0, %1) step(64, 32) {
+          "use0"(%arg0, %arg1) : (index, index) -> ()
+          scf.forall.in_parallel {}
+        } {mapping = [#iree_codegen.workgroup_mapping<y>, #iree_codegen.workgroup_mapping<x>]}
+        return
+      }
+      func.func @entry_point1() {
+        %cst0 = hal.interface.constant.load layout(#pipeline_layout) ordinal(0) : index
+        %cst1 = hal.interface.constant.load layout(#pipeline_layout) ordinal(1) : index
+        %cst2 = hal.interface.constant.load layout(#pipeline_layout) ordinal(2) : index
+        %0 = iree_tensor_ext.dispatch.workload.ordinal %cst0, 0 : index
+        %1 = iree_tensor_ext.dispatch.workload.ordinal %cst1, 1 : index
+        %2 = iree_tensor_ext.dispatch.workload.ordinal %cst2, 2 : index
+        scf.forall (%arg0, %arg1) = (0, 0) to (%0, %1) step(64, 32) {
+          "use1"(%arg0, %arg1) : (index, index) -> ()
+          scf.forall.in_parallel {}
+        } {mapping = [#iree_codegen.workgroup_mapping<y>, #iree_codegen.workgroup_mapping<x>]}
+        return
+      }
+    }
+  }
+}
+//  CHECK-DAG: #[[MAP0:.+]] = affine_map<()[s0] -> (s0 ceildiv 64)
+//  CHECK-DAG: #[[MAP1:.+]] = affine_map<()[s0] -> (s0 ceildiv 32)
+//      CHECK: hal.executable.export public @entry_point0 layout
+// CHECK-SAME:     %[[ARG1:[a-zA-z0-9]+]]: index
+// CHECK-SAME:     %[[ARG2:[a-zA-z0-9]+]]: index
+// CHECK-SAME:     %[[ARG3:[a-zA-z0-9]+]]: index
+//  CHECK-DAG:   %[[WG_Y:.+]] = affine.apply #[[MAP0]]()[%[[ARG1]]]
+//  CHECK-DAG:   %[[WG_X:.+]] = affine.apply #[[MAP1]]()[%[[ARG2]]]
+//  CHECK-DAG:   %[[C1:.+]] = arith.constant 1 : index
+//      CHECK:   hal.return %[[WG_X]], %[[WG_Y]], %[[C1]]
+//      CHECK: hal.executable.export public @entry_point1 layout
+// CHECK-SAME:     %[[ARG1_1:[a-zA-z0-9]+]]: index
+// CHECK-SAME:     %[[ARG2_1:[a-zA-z0-9]+]]: index
+// CHECK-SAME:     %[[ARG3_1:[a-zA-z0-9]+]]: index
+//  CHECK-DAG:   %[[WG_Y_1:.+]] = affine.apply #[[MAP0]]()[%[[ARG1_1]]]
+//  CHECK-DAG:   %[[WG_X_1:.+]] = affine.apply #[[MAP1]]()[%[[ARG2_1]]]
+//  CHECK-DAG:   %[[C1_1:.+]] = arith.constant 1 : index
+//      CHECK:   hal.return %[[WG_X_1]], %[[WG_Y_1]], %[[C1_1]]
```

Comment:
I'd remove the type (i.e., ` : index`) from checks because they are not important.

> Tests should be minimal, and only check what is absolutely necessary.

https://mlir.llvm.org/getting_started/TestingGuide/#filecheck-best-practices

Reply:
I kept the `: index` on the input arguments since they improve readability to me, and I've had a number of issues with matching multiple values on a `CHECK-SAME` line without being careful about the regex.

In general I try to avoid matching the operand types of an op and only check result or block arg types to keep tests simple like you're suggesting though.